File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def lowestCommonAncestor(self, root: TreeNode, p: TreeNode, q: TreeNode) -> TreeNode:
        if q.val < p.val:
            q, p = p,q
        
        while root:
            if root.val > p.val and root.val < q.val:
                return root
            if root.val == p.val or root.val == q.val:
                return root
            if root.val > p.val and root.val > q.val:
                root = root.left
            if root.val < p.val and root.val < q.val:
                root = root.right
            
        

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        Queue<TreeNode> queue = new LinkedList<>();
        List<List<Integer>> res = new ArrayList<>();
        if (root != null){
            queue.add(root);
        }
        int level = 0;
        while(queue.size() > 0 ){
            List<Integer> list = new ArrayList<>();
            int levelLength = queue.size();
            for (int i = 0; i < levelLength; i++){
                TreeNode cur = queue.remove();
                list.add(cur.val);
                if (cur.left != null){
                    queue.add(cur.left);
                }
                if (cur.right != null){
                    queue.add(cur.right);
                }
            }
            level++;
            res.add(list);
        }
        return res;
    }
}
